cmd/root.go

```diff
@@ -120,29 +111,44 @@ or ./scorecard --{npm,pypi,rubgems}=<package_name> [--checks=check1,...] [--show
 		} else {
 			enabledChecks = checks.AllChecks
 		}
-		for checkName := range enabledChecks {
-			if format == formatDefault {
+		if format == formatDefault {
+			for checkName := range enabledChecks {
 				fmt.Fprintf(os.Stderr, "Starting [%s]\n", checkName)
 			}
 		}
 		ctx := context.Background()
 
-		resultsCh := pkg.RunScorecards(ctx, sugar, repo, enabledChecks)
-		// Collect results
-		results := []checker.CheckResult{}
-		for result := range resultsCh {
-			if format == formatDefault {
-				fmt.Fprintf(os.Stderr, "Finished [%s]\n", result.Name)
-			}
-			results = append(results, result)
-		}
+		repoResult := pkg.RunScorecards(ctx, sugar, repo, enabledChecks)
+		repoResult.Metadata = append(repoResult.Metadata, metaData...)
 
 		// Sort them by name
-		sort.Slice(results, func(i, j int) bool {
-			return results[i].Name < results[j].Name
+		sort.Slice(repoResult.CheckResults, func(i, j int) bool {
+			return repoResult.CheckResults[i].Name < repoResult.CheckResults[j].Name
 		})
 
-		outputFn(results)
+		if format == formatDefault {
+			for checkName := range enabledChecks {
+				fmt.Fprintf(os.Stderr, "Finished [%s]\n", checkName)
+			}
+			fmt.Println("\nRESULTS\n-------")
+		}
+
+		switch format {
+		case formatDefault:
+			if err := repoResult.AsString(showDetails, os.Stdout); err != nil {
+				panic(err)
```

Comment:
nit: can save some duplication if we do

```go
var err error
switch format {
case formatDefault:
  err = repoResult.AsString(...)
case formatCSV:
  err = repoResult.AsCSV(...)
...

if err != nil {
  log.Fatalf("Failed to output results: %w", err)
}
```

(let's also be consistent and use Fatalf here).

Reply:
Done.